src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -712,22 +708,52 @@ bool EmitAnchors()
                 return false;
             }
 
-            // Emits a case-sensitive prefix search for a string at the beginning of the pattern.
-            void EmitIndexOf_LeftToRight(string prefix)
+            // Emits a case-sensitive left-to-right search for a substring.
+            void EmitIndexOf_LeftToRight()
             {
-                writer.WriteLine($"// The pattern begins with a literal {Literal(prefix)}. Find the next occurrence.");
+                RegexFindOptimizations opts = regexTree.FindOptimizations;
+
+                string substring = "";
+                string offset = "";
```

Comment:
NIT: I know this is compile time so doesn't matter much but should these two be `string.Empty` instead?

Reply:
I don't have a strong opinion.  We use both throughout dotnet/runtime.  I'll leave it as is.